packages/viewer/src/commands/streamCommands.ts

```diff
@@ -47,7 +49,133 @@ export function startStream(
   };
 }
 
+export function createSceneAlteration(
+  sceneViewId: UUID.UUID,
+  query: QueryExpression,
+  operations: OperationDefinition[]
+): Command<Promise<vertexvis.protobuf.stream.IStreamResponse>> {
+  return ({ stream }: CommandContext) => {
+    const pbOperations: vertexvis.protobuf.stream.ISceneOperation[] = [
+      buildSceneOperation(query, operations),
+    ];
+    const request: vertexvis.protobuf.stream.ICreateSceneAlterationRequest = {
+      sceneViewId: new vertexvis.protobuf.core.Uuid({
+        hex: sceneViewId,
+      }),
+      operations: pbOperations,
+    };
+
+    return stream.createSceneAlteration(request);
+  };
+}
+
+function buildSceneOperation(
+  query: QueryExpression,
+  operations: OperationDefinition[]
+): vertexvis.protobuf.stream.ISceneOperation {
+  const operationTypes: vertexvis.protobuf.stream.IOperationType[] = buildOperationTypes(
+    operations
+  );
+
+  switch (query.type) {
+    case 'and':
+      return {
+        and: buildQueryCollection(query),
+        operationTypes,
+      };
+    case 'or':
+      return {
+        or: buildQueryCollection(query),
+        operationTypes,
+      };
+    case 'item-id':
+    case 'supplied-id':
+      return {
+        item: {
+          sceneItemQuery: buildSceneItemQuery(query),
+        },
+        operationTypes,
+      };
+    default:
+      return {};
+  }
```

Comment:
This can be simplified

```ts
function buildSceneOperation(
  query: QueryExpression,
  operations: OperationDefinition[]
): vertexvis.protobuf.stream.ISceneOperation {
  const operationTypes = buildOperationTypes(operations);

  switch (query.type) {
    case 'and':
    case 'or':
      return {
        [query.type]: {
          queries: query.expressions.map(exp => ({
            sceneItemQuery: buildSceneItemQuery(exp),
          })),
          operationTypes,
        },
      };
    case 'item-id':
    case 'supplied-id':
      return {
        item: {
          sceneItemQuery: buildSceneItemQuery(query),
        },
        operationTypes,
      };
    default:
      return {};
  }
}
```